library/src/include/utility.h

```diff
@@ -59,6 +59,33 @@ __forceinline__ __device__ __host__ T load_scalar(const T* xp)
     return *xp;
 }
 
+// Load a pointer from a batch. If the argument is a T**, use block to index it and
+// add the offset, if the argument is a T*, add block * stride to pointer and add offset.
+
+// For device pointers
+template <typename T>
+__forceinline__ __device__ __host__ T*
+                                    load_ptr_batch(T* p, rocblas_int block, rocblas_int offset, rocblas_stride stride)
+{
+    return p + block * stride + offset;
+}
+
+// For device array of device pointers
+template <typename T>
+__forceinline__ __device__ __host__ T*
+                                    load_ptr_batch(T* const* p, rocblas_int block, rocblas_int offset, rocblas_stride stride)
+{
+    return p[block] + offset;
+}
+
+// For device array of device pointers
+template <typename T>
+__forceinline__ __device__ __host__ T*
+                                    load_ptr_batch(T** p, rocblas_int block, rocblas_int offset, rocblas_stride stride)
+{
+    return p[block] + offset;
+}
+
```

Comment:
Why was this moved above `load_scalar`'s overload for `rocblas_half2`?

Reply:
Must have happened in a merge, rearranged stuff in 41463e5.